src/main/resources/mapping.json

```diff
@@ -36,15 +36,24 @@
           "description": "XPath to the data for the field to map"
         },
         "translation": {
-          "type": "string",
+          "enum" : ["toDouble", "toInteger", "lookupMaterialTypeId", "lookupLocationId", "lookupVendorId", "getPurchaseOptionCode"],
           "description": "Method to translate the specified type"
         },
         "combinator": {
-          "type": "string",
+          "enum": ["concat", "multiply"],
           "description": "Method to combine the specified type with other values"
         },
-        "default": {
-          "type": "string"
+        "default" : {
+          "type" : "string",
+          "description" : "default value"
+        },
+        "defaultMapping" : {
+          "$ref" : "#",
+          "description" : "default value found from the mapping of another node"
+        },
+        "fromOtherField" : {  //in the config json, the order of dataSource is important as this field refers to previous dataSource objects.
```

Comment:
I think the comment makes this invalid json.  Maybe move this to the description?